File: 453-Minimum Moves to Equal Array Elements-最小操作次数使数组元素相等/453-Minimum Moves to Equal Array Elements-最小操作次数使数组元素相等.js
/**
 * @param {number[]} nums
 * @return {number}
 * 逆向思维太重要了，n-1个元素+1 = 1个元素减1
 */
var minMoves = function(nums) {
    let res = 0
    let min = Math.min(...nums)
    let len = nums.length
    for(let i =0; i < len; i++){
        res += (nums[i] - min)
    }
    return res
};

File: 453-Minimum Moves to Equal Array Elements-最小操作次数使数组元素相等/453-Minimum Moves to Equal Array Elements-最小操作次数使数组元素相等.java
class Solution {
    public int minMoves(int[] nums) {
        int res = 0;
        int min = Arrays.stream(nums).min().getAsInt();
        int len = nums.length;
        for(int i =0; i < len; i++){
            res += (nums[i] - min);
        }
        return res;
    }
}
